src/hooks/useDebounce.ts

```diff
@@ -0,0 +1,19 @@
+import {useState, useEffect} from 'react';
+
+const useDebounce = (value: string, delay: number) => {
+  const [debouncedValue, setDebouncedValue] = useState(value);
```

Comment:
```suggestion
const useDebounce = <T>(value: T, delay: number) => {
  const [debouncedValue, setDebouncedValue] = useState(value);
```

debounce할 때 value값이 제너릭이면 더 좋을 것 같습니다